src/querydb/store.rs

```diff
@@ -0,0 +1,250 @@
+use crate::querydb::prelude::*;
+use pallas::ledger::traverse::{MultiEraBlock, MultiEraPolicyAssets, MultiEraTx};
+use redb::{
+    Database, MultimapTable, ReadOnlyMultimapTable, ReadOnlyTable, ReadTransaction,
+    ReadableMultimapTable, ReadableTable, Table, WriteTransaction,
+};
+use std::{ops::Deref, path::Path};
+
+pub struct Store {
+    inner_store: Database,
+}
+
+impl Store {
+    pub fn new(path: impl AsRef<Path>) -> Result<Self, StoreError> {
+        Ok(Store {
+            inner_store: Database::create(path).map_err(|e| StoreError::ReDBError(Box::new(e)))?,
+        })
+    }
+    pub fn apply_block(&self, block_cbor: BlockValueType) -> Result<(), StoreError> {
+        let write_tx: WriteTransaction = self
+            .inner_store
+            .begin_write()
+            .map_err(|e| StoreError::ReDBError(Box::new(e)))?;
+        let block: MultiEraBlock = self.store_block(&write_tx, block_cbor)?;
+        self.store_txs(&write_tx, &block)?;
+        write_tx
+            .commit()
+            .map_err(|e| StoreError::ReDBError(Box::new(e)))?;
+        Ok(())
+    }
+
+    fn store_block<'a>(
+        &self,
+        write_tx: &'a WriteTransaction,
+        block_cbor: &'a [u8],
+    ) -> Result<MultiEraBlock<'a>, StoreError> {
+        let block: MultiEraBlock =
+            MultiEraBlock::decode(block_cbor).map_err(StoreError::BlockDecoding)?;
+        let mut block_table: Table<BlockKeyType, BlockValueType> = write_tx
+            .open_table(BLOCK_TABLE)
+            .map_err(|e| StoreError::ReDBError(Box::new(e)))?;
+        let _ = block_table.insert(block.hash().deref(), block_cbor);
+        let mut blockchain_tip_table: Table<ChainTipKeyType, ChainTipValueType> = write_tx
+            .open_table(CHAIN_TIP_TABLE)
+            .map_err(|e| StoreError::ReDBError(Box::new(e)))?;
+        let _ = blockchain_tip_table.insert((), block.hash().deref());
```

Comment:
we don't need a `CHAIN_TIP_TABLE`. There's nothing we need to store regarding the chain tip. We can just query the last entry from the `BLOCK_TABLE`.

Reply:
The problem is blocks are indexed by hash in `BLOCK_TABLE`. But the order between block hashes doesn't coincide with the order in which blocks are inserted through time. And tables in ReDB are ordered by key..

The current approach may be a bit unattractive from a design standpoint, but shouldn't spend too much memory or CPU anyways...